src/main/java/eu/miaplatform/decorators/predecorators/PreDecoratorRequest.java

```diff
@@ -0,0 +1,96 @@
+package eu.miaplatform.decorators.predecorators;
+
+import eu.miaplatform.decorators.DecoratorRequest;
+import eu.miaplatform.service.EnvConfiguration;
+import eu.miaplatform.service.EnvConfigurationException;
+import lombok.*;
+
+import java.util.Map;
+
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@Data
+public class PreDecoratorRequest {
+    private String method;
+    private String path;
+    private Map<String, String> headers;
+    private Map<String, String> query;
+    private Object body;
+
+    public PreDecoratorRequestProxy.Builder changeOriginalRequest() {
+        return new PreDecoratorRequestProxy.Builder(
+                DecoratorRequest.builder()
+                        .method(this.getMethod())
+                        .path(this.getPath())
+                        .headers(this.getHeaders())
+                        .query(this.getQuery())
+                        .body(this.getBody())
+                        .build());
+    }
+
+    public PreDecoratorRequest leaveOriginalRequestUnmodified() {
+        return null;
+    }
+
+    public DecoratorRequest getRequest() {
+        return DecoratorRequest.builder()
+                .method(this.method)
+                .path(this.path)
+                .headers(this.headers)
+                .query(this.query)
+                .body(this.body)
+                .build();
+    }
+
+    public String getOriginalRequestPath() {
+        return this.getPath();
+    }
+
+    public String getOriginalRequestMethod() {
+        return this.getMethod();
+    }
+
+    public Map<String, String> getOriginalRequestHeaders() {
+        return this.getHeaders();
+    }
+
+    public Map<String, String> getOriginalRequestQuery() {
+        return this.getQuery();
+    }
+
+    public Object getOriginalRequestBody() {
+        return this.getBody();
+    }
+
+    public String getUserId() {
+        try  {
+            return this.headers.get(EnvConfiguration.get("USERID_HEADER_KEY"));
+        } catch(EnvConfigurationException ex) {
```

Comment:
All these silenced exception are bad, the developer will never know that some environment is missing.

The developer should instead design its env schema, and invoke the parseEnvironment as soon as possible to check that the env is properly configured. After that these function can rely the fact that using EnvConfiguration.get will return the proper value (the one set in the environment or the default set by the developer)